src/APIM_ARMTemplate/apimtemplate/Common/Templates/Policy/PolicyTemplateUtils.cs

```diff
@@ -0,0 +1,38 @@
+﻿using apimtemplate.Extractor.Models;
+using Microsoft.Extensions.Caching.Memory;
+using System.IO;
+
+namespace apimtemplate.Common.Templates.Policy
+{
+    internal static class PolicyTemplateUtils
+    {
+        private static readonly IMemoryCache _policyCache = new MemoryCache(new MemoryCacheOptions());
+
+        public static string GetPolicyContent(ExtractorParameters extractorParameters, PolicyTemplateResource policyTemplateResource)
+        {
+            // the backend is used in a policy if the xml contains a set-backend-service policy, which will reference the backend's url or id
+            string policyContent = policyTemplateResource.properties.value;
+
+            // check if this is a file or is it the raw policy content
+            if (policyContent.Contains(".xml"))
+            {
+                var key = policyContent;
+                //check cache
+                if (_policyCache.TryGetValue(key, out string content))
+                    return content;
```

Comment:
We typically use this notation
```suggestion
                if (_policyCache.TryGetValue(key, out string content))
                {
                    return content;
                }
```

Reply:
fixed style, thanks